src/core/Plugin.cpp

```diff
@@ -74,6 +80,97 @@ Plugin::~Plugin()
 
 
 
+template<class T>
+T use_this_or(T this_param, T or_param)
+{
+	return this_param ? this_param : or_param;
+}
+
+
+
+
+QString use_this_or(QString this_param, QString or_param)
+{
+	return this_param.isNull() ? or_param : this_param;
+}
+
+
+
+
+QString Plugin::displayName() const
+{
+	return Model::displayName().isEmpty() // currently always empty
+		? (m_descriptor->subPluginFeatures && m_key.isValid())
+			// get from sub plugin
+			? m_key.displayName()
+			// get from plugin
+			: m_descriptor->displayName
+		: Model::displayName();
+}
+
+
+
+
+const PixmapLoader* Plugin::logo() const
+{
+	return (m_descriptor->subPluginFeatures && m_key.isValid())
+		? m_key.logo()
+		: m_descriptor->logo;
+}
+
+
+
+
+QString Plugin::Descriptor::SubPluginFeatures::Key::additionalFileExtensions() const
+{
+	Q_ASSERT(isValid());
+	return desc->subPluginFeatures
+		// get from sub plugin
+		? desc->subPluginFeatures->additionalFileExtensions(*this)
+		// get from plugin
+		: nullptr;
```

Comment:
`nullptr` is `get from plugin` (maybe `QString()` instead)?

Reply:
You are right in both cases. I removed "get from plugin", because obviously, you can't get additional file extensions if there is no sub plugin. Also I changed `nullptr` to `QString`.